core/core/src/sf_core/json_schema_validator.rs

```diff
@@ -183,5 +183,17 @@ mod test {
             .unwrap(),
         );
         assert!(result.is_err());
+
+        let result = security_validator.validate(
+            &HostValue::deserialize(&json!({
+                "security_config": {
+                    "partial_basic": {
+                        "username": "username"
+                    }
+                }
+            }))
+            .unwrap(),
+        );
+        assert!(result.is_err());
```

Comment:
issue: I think we should also test this, and decide what behavior we want - I think we want to reject this.
```
{
  "some_name": {} // empty object
}
```